src/main/java/com/aws/greengrass/componentmanager/ClientConfigurationUtils.java

```diff
@@ -34,11 +38,38 @@
 
 public final class ClientConfigurationUtils {
     private static final Logger logger = LogManager.getLogger(ClientConfigurationUtils.class);
+    private static final RetryUtils.RetryConfig infiniteAttemptsRetryConfig =
+            RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofSeconds(30L))
+                    .maxRetryInterval(Duration.ofMinutes(45L)).maxAttempt(Integer.MAX_VALUE)
+                    .retryableExceptions(Collections.singletonList(RetryableException.class)).build();
 
     // Utility class. These methods are used by plugins so this interface *must not* change
     private ClientConfigurationUtils() {
     }
 
+    /**
+     * Configure the http client builder with the required certificates for the mutual auth connection.
+     *
+     * @param deviceConfiguration {@link DeviceConfiguration}
+     * @return configured http client
+     * @throws TLSAuthException if there is an issue configuring the http client
+     */
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    public static ApacheHttpClient.Builder getConfiguredClientBuilder(DeviceConfiguration deviceConfiguration)
+            throws TLSAuthException {
```

Comment:
Are there other plugins who use this function? Would this break old versions of other plugins?

Reply:
No other plugin in the GG repository is using this.